WinUIGallery/DataModel/ControlInfoDataSource.cs

```diff
@@ -194,105 +197,70 @@ public async Task<ControlInfoDataGroup> GetGroupFromItemAsync(string uniqueId)
 
         private async Task GetControlInfoDataAsync()
         {
-            lock (_lock)
+            if (this.Groups.Count() != 0)
             {
-                if (this.Groups.Count() != 0)
-                {
-                    return;
-                }
+                return;
             }
 
-            string jsonText = await FileLoader.LoadText("DataModel/ControlInfoData.json");
-
-            JsonObject jsonObject = JsonObject.Parse(jsonText);
-            JsonArray jsonArray = jsonObject["Groups"].GetArray();
-
-            lock (_lock)
+            var jsonFile = await FileLoader.GetJsonFile("DataModel/ControlInfoData.json");
+            using FileStream openStream = File.OpenRead(jsonFile.Path);
+            var controlInfoDataGroup = await JsonSerializer.DeserializeAsync<Root>(openStream, new JsonSerializerOptions
+            {
+                WriteIndented = true,
+                PropertyNameCaseInsensitive = true
+            });
+            string pageRoot = "AppUIBasics.ControlPages.";
+            foreach (var groupItem in controlInfoDataGroup.Groups)
             {
-                string pageRoot = "AppUIBasics.ControlPages.";
-                foreach (JsonValue groupValue in jsonArray)
+                ControlInfoDataGroup group = new ControlInfoDataGroup(groupItem.UniqueId,
```

Comment:
Why do we still need to this, doesn JSONSerializer support parsing nested trees?

Reply:
I tried not to touch the original code, I will check

Reply:
If you need a reference for this, you can look at the IconsDataSource which just lets JSONSerializer handle everything.

Reply:
Did it get better?